src/main/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisRecordProcessor.kt

```
@@ -66,6 +68,12 @@ class AwsKinesisRecordProcessor(
     private fun processRecord(recordData: String) {
         log.debug("Processing record.")
         val message = recordMapper.deserializeFor(recordData, handler)
+        val violations = validator?.validate(message) ?: setOf()
+        if (violations.isNotEmpty()) {
+            log.warn("skip invalid record with violations: $violations message: $message")
```

I would throw an exception. If you do so, we would automatically run into the log statement in line 51. So all kinds of "failings" would be handled there in a single place.

---

fixed